src/Identifier/Ldap/ExtensionAdapter.php

```diff
@@ -89,15 +89,22 @@ public function getConnection()
      * @param string $host Hostname
      * @param int $port Port
      * @param array $options Additonal LDAP options
+     * @param bool $tls if to try starting TLS
      * @return void
      */
-    public function connect(string $host, int $port, array $options): void
+    public function connect(string $host, int $port, array $options, bool $tls = false): void
```

Comment:
The signature change makes this a non backwards compatible change and require a major version bump.

Couldn't we instead use `tls` key in `$options` instead and unset it before `$options` are looped over for `setOption()` calls?

Reply:
Is it not backwards-compatible if it defaults to false? 

I did consider adding it to the options array (or just checking to see if any _TLS_ options had been set) but it seemed clumsy.

Reply:
> Is it not backwards-compatible if it defaults to false?

For a user it is, but since the interface's signature changed any custom implementation of it would break.

Reply:
Fair enough, although I still think that it is pretty clumsy

Reply:
Done